signatures.md

```diff
@@ -52,163 +49,350 @@ Note: `Message` is used here as the object being signed, but this interface shou
+        //   err - a standard error message indicating any process issues
+        // In:
+  			//   smsg - a SignedMessage which contains {Message, Address, SignatureBytes}
+        //   pk - a Public Key of the appropriate type
+        //
+        Verify(smsg SignedMessgage, pk PublicKey) (valid bool, err error)
+  
+  			// VerifyAggregate validates a collection of statements that are represeted by 
+  			// a signature aggregate: "For each M_i and PK_i in smsg, only this collection 
+  			// of {M}, {PK}, and signers could have generated `smsg.SignatureBytes`.
+  			//
+        // Out:
+        //   valid - a boolean value indicating the signature is valid
+        //   err - a standard error message indicating any process issues
         // In:
-        //    m - a series of bytes representing the signed message
-        //    pk - the public key belonging to the signer `M`
-        //    sig - a series of bytes representing a signature usually `r`|`s`
+        //   smsg - a SignedMessage which contains {[]Message, []Address, SignatureBytes}
+        //   pks - an array of Public Keys of the appropriate type associated with a message
+  			// 	 at index `i` in []Message.
         //
-        Verify(m Messgage, pk PublicKey, sig SignatureBytes) (valid bool, err error)
+  			VerifyAggregate(smsg SignedMessgage,  pks []PublicKeys) (valid bool, err error)
 
-        // Recover, as its name implies, recovers a public key associated with a 
-        // particular signature. In the case of ECDSA signatures, this function can 
-        // be fulfilled via the 'ECRecover' method. If a different signature scheme 
-        // is used, then some other mechanism of 'recovering' a message authors 
-        // public key must be provided.
+        // Recover determines the public key associated with a particular signature. For
+  			// ECDSA signatures public keys can be recovered from SignatureBytes and an
+  			// associated message. For BLS signatures, public keys will be recovered by lookup 
+  			// using a mapping from Address to public key.
         //
         // Out:
-        //    pk - the public key associated with `M` who signed `m`
+        //    pk - the public key associated with `M` who signed `msg`
         //    err - a standard error message indicating any process issues
         //    **
         // In:
-        //    m - a series of bytes representing the signed message
-        //    sig - a series of bytes representing a signature usually `r`|`s`
+        //    msg - a series of bytes representing the signed message
+        //    sig - a series of bytes representing a signature
         //
-        Recover(m Message, sig SignatureBytes) (pk PublicKey, err error)
+        Recover(msg Message, addr Address, sig SignatureBytes) (pk PublicKey, err error)
 }
 ```
 
+### Signed Message
+
+This is the container for a signature that combines a message with an address and signature. 
```

Comment:
I'm confused by the plurality of addresses and messages in the structure given this description.

I understand the use case for multiple addresses, but what is the case for multiple different messages? This seems like redundant data in the case of multiple signers of an identical message.

A brief outline of use cases and capabilities at the top of this doc would help greatly if my understanding of the motivations and context is lacking.

Reply:
I'm note sure I quite understand but I've reworked the description. Agree it is very confusing. Will push. Let me know if that answers your questions.

Reply:
I still don't understand why there are multiple messages in the container. The comment uses the singular "a message and an address" which disagrees with the code directly below. Please clarify the intended role of a `SignedMessage`. E.g. you could write: 

```A container for one or multiple messages and an aggregation of the signatures of those messages by corresponding addresses. In the case of a single message and address, the signature is a vanilla signature, not an aggregate. In the case of multiple messages, the signature is assumed to be an aggregate. See *Determining signature type* below for more detail.```

But I don't know whether that is in fact correct or complete.